### PR TITLE
NXP: LPC: various cleanups

### LINKED ISSUE #24439: LPCXpresso55S69_ns target : build failed
There are some compilation problems when trying to compile Zephyr samples with the target "lpcxpresso55s69_ns".

**To Reproduce**
`west build -p auto -b lpcxpresso55s69_ns samples/hello_world`

**Issues**
It seems that there are 3 problems : 
One in `lpcxpresso55s69_ns_defconfig` file : CONFIG_ARM is not user configurable
and two in `nxp_lpc55s6x_ns.dtsi` file : It's missing i2c.h as included file and we have to replace kinetis-gpio by lpc-gpio otherwise the declaration of kinetis-port is missing.

**Environment :**
 - OS: Ubuntu
 - Toolchain : Zephyr SDK
 - Zephyr version : 2.2.99

### PR DESCRIPTION
**Pull request by galak**
* cleanup and refactoring of LPC55S6x and LPC54xxx device trees
* Remove stale Kconfig options related to LPC support
* Get lpcxpresso55s69_ns building
* Move to nodelabel's for pinmux enablement

### CODE REVIEW

File: dts/arm/nxp/nxp_lpc55S6x_common.dtsi [hunk 1]
```diff
@@ -204,12 +204,14 @@
 		status = "disabled";
 	};
 
-	flexcomm8: flexcomm@9f000 {
-		compatible = "nxp,lpc-flexcomm";
+	hs_lspi: spi@9f000 {
```
Comment (MaureenHelm) [inline dts/arm/nxp/nxp_lpc55S6x_common.dtsi:189]: where did the "l" come from?

File: dts/arm/nxp/nxp_lpc55S6x_common.dtsi [hunk 2]
```diff
@@ -204,12 +204,14 @@
...
+		compatible = "nxp,lpc-spi";
 		reg = <0x9f000 0x1000>;
 		interrupts = <59 0>;
-		label = "FLEXCOMM_8";
+		label = "HS_LSPI";
```
Comment (MaureenHelm) [inline dts/arm/nxp/nxp_lpc55S6x_common.dtsi:193]: Need to update `tests/drivers/spi/spi_loopback/boards/lpcxpresso55s69_cpu0.conf` too

File: boards/arm/lpcxpresso54114/pinmux.c [hunk 3]
```diff
@@ -23,7 +23,7 @@ static int lpcxpresso_54114_pinmux_init(struct device *dev)
 		device_get_binding(CONFIG_PINMUX_MCUX_LPC_PORT1_NAME);
 #endif
 
-#ifdef CONFIG_UART_MCUX_FLEXCOMM_0
+#if DT_HAS_NODE(DT_NODELABEL(flexcomm0)) && defined(CONFIG_UART_MCUX_FLEXCOMM)
```
Comment (MaureenHelm) [inline boards/arm/lpcxpresso54114/pinmux.c:26]: is there a macro to check the compat on flexcomm0?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: dts/arm/nxp/nxp_lpc55S6x_common.dtsi (hunk 1 above)
Comment (galak) [reply dts/arm/nxp/nxp_lpc55S6x_common.dtsi:189]: its what is used in the NXP HAL.  hs_lspi/hs_spi seem to be intermixed.

File: boards/arm/lpcxpresso54114/pinmux.c (hunk 3 above)
Comment (galak) [reply boards/arm/lpcxpresso54114/pinmux.c:26]: `DT_NODE_HAS_COMPAT(DT_NODELABEL(flexcomm0), nxp_lpc_usart)`

File: dts/arm/nxp/nxp_lpc55S6x_common.dtsi (hunk 1 above)
Comment (galak) [reply dts/arm/nxp/nxp_lpc55S6x_common.dtsi:189]: https://github.com/zephyrproject-rtos/hal_nxp/blob/87850e41cd48e4472adbae03c9623ea349255d48/mcux/devices/LPC55S69/fsl_clock.h#L258

File: dts/arm/nxp/nxp_lpc55S6x_common.dtsi (hunk 2 above)
Comment (galak) [reply dts/arm/nxp/nxp_lpc55S6x_common.dtsi:193]: fixed

File: boards/arm/lpcxpresso54114/pinmux.c (hunk 3 above)
Comment (MaureenHelm) [reply boards/arm/lpcxpresso54114/pinmux.c:26]: ok, then we should use it here and everywhere else we set up flexcomm pins

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: boards/arm/lpcxpresso54114/pinmux.c (hunk 3 above)
Comment (galak) [reply boards/arm/lpcxpresso54114/pinmux.c]: Done

**Comment by galak** [comment]
Fixes #24439

**Review by henrikbrixandersen — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**